tests/test_coordinateConverter.py

```diff
@@ -0,0 +1,556 @@
+                self.assertPairsAlmostEqual(initialBeamInfo.focalPlanePos, focalPlanePos,
+                                            maxDiff=self.maxFocalPlanePosErr)
+                self.cco.offsetFocalPlanePos(pupilOffset=pupilOffset, focalPlaneOffset=focalPlaneOffset,
+                                             beam=beam)
+                finalBeamInfo = self.cco[beam]
+                if pupilOffset is None:
+                    desiredPupilPos = pupilPos
+                else:
+                    desiredPupilPos = np.add(pupilPos, pupilOffset)
+                if focalPlaneOffset is None:
+                    desiredFocalPlanePos = focalPlanePos
+                else:
+                    desiredFocalPlanePos = np.add(focalPlanePos, focalPlaneOffset)
+                self.assertPairsAlmostEqual(finalBeamInfo.pupilPos, desiredPupilPos, self.maxPupilPosErr)
+                self.assertPairsAlmostEqual(finalBeamInfo.focalPlanePos, desiredFocalPlanePos,
+                                            maxDiff=self.maxFocalPlanePosErr)
+
+    def testSampleBasics(self):
+        """Test basic elements of the sample coordinate converter
+        """
+        self.assertEqual(len(self.cco.cameraGeom), len(self.scc.detectorFracPosList))
+        self.assertEqual(list(self.cco.cameraGeom.getNameIter()),
+                         ["D" + str(i) for i in range(len(self.scc.detectorFracPosList))])
+        self.assertEqual(len(self.cco), 6)
+        self.assertEqual(tuple(self.cco.beamNames), ("0", "1", "2", "3", "4", "5"))
+        self.assertEqual(self.cco.maskInfo.numHoles, 6)
+        self.assertPairsAlmostEqual(self.cco.maskInfo.getHolePos("0"), (0, 0))
+        for beamInfo, beam in zip(self.cco, self.cco.beamNames):
+            self.assertEqual(beamInfo.name, beam)
+
+    def testSetPupilFieldAngleTrivial(self):
+        """Test setPupilFieldAngle for the trivial case of hole 0
+        aimed perpendicular to the center of the pupil
+        """
+        self.cco.setPupilFieldAngle(pupilPos=(0, 0))
+
+        # the telescope should be pointed at the center of the CBP and vice-versa
+        # NOTE: It would be nice to get better than the 0.0028" that I measure
+        self.assertSpherePointsAlmostEqual(self.cco.telAzAltInternal,
+                                           SpherePoint(Vector3d(*self.cco.config.cbpPosition)),
+                                           maxSep=0.01*arcseconds)
+        self.assertSpherePointsAlmostEqual(self.cco.cbpAzAltInternal,
+                                           SpherePoint(Vector3d(*(-self.cco.config.cbpPosition))),
+                                           maxSep=0.01*arcseconds)
+
+        self.assertAnglesAlmostEqual(self.cco.telRotInternal, 0*degrees)
+
+        # beam 0 should be pointed to the center of the pupil, normal to the pupil
+        # and land on the center of the focal plane and the center of detector D0
```

Comment:
I'm not sure what you are asking for here. The comment currently says "and land on the center of the focal plane and the center of detector D0". Are you asking for an "at" after the "and" of "and the center of detector D0" or something more.

Reply:
It's really unimportant, I shouldn't have said anything. I just paused to contemplate this when I read it, because (assuming it's centered, which I'm pretty sure it is), pedantically speaking, the center of the focal plane is the center of that chip, so it's kind of being said twice here, and so the second part confused me momentarily. Just leave it, it's fine, like I said, I shouldn't have said anything!

Reply:
That makes sense. I'll change it to "and land on the center of the focal plane, which is also the center of detector D0"